File: twoSum.js
function twoSum(nums, target) {
    let m = new Map()
    for(let i = 0; i < nums.length; i++){
        if(m.has(target-nums[i]))
            return [m.get(target-nums[i]), i]
        m.set(nums[i], i)
    }
}

File: two_sum.py
from typing import List 
def two_sum(nums: List[int], target: int) -> List[int]:
    indices = {}
    for i in range(len(nums)):
        if target - nums[i] in indices:
            return [indices[target - nums[i]], i]
        indices[nums[i]] = i
    return []

print(twoSum([0, 3, 2, 5, 1], 4), "[1, 4]")
